src/main/java/com/codecool/shop/controller/AddController.java

```diff
@@ -0,0 +1,103 @@
+package com.codecool.shop.controller;
+
+import com.codecool.shop.dao.implementation.ProductDaoMem;
+import com.codecool.shop.model.Cart;
+import com.codecool.shop.model.Product;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+@WebServlet(urlPatterns = {"/add-product"})
+public class AddController extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+
+        int productId = Integer.parseInt(req.getParameter("id"));
+        ProductDaoMem productDaoMem = ProductDaoMem.getInstance();
+        List<Product> products = productDaoMem.getAll();
+        checkGetMethodParameters(req, productId, products);
+
+    }
+    /**
+     *Choose to change quantity (add or remove) or add a new one (or increase the quantity).
+     *
+     * @param request - HttpServletRequest.
+     * @param productId - The id of the product to change quantity.
+     * @param products - Product in store.
+     */
+    private void checkGetMethodParameters(HttpServletRequest request, int productId, List<Product> products) {
```

Comment:
I would modify it like follows:
* Pass only the `quantity` parameter's value instead of the full request because only that info is used
* rename the method accordingly
Actually I'm wondering: do you really need to pass the `products` list as well? Why don't the concrete method calling in when it used?